### PR TITLE
DOC: Fix capitalization of the word pandas in the docs

### LINKED ISSUE #32316: DOC: Standardize references to pandas in the documentation
In our documentation (files in `doc/source/../*.rst`), we refer to pandas as `Pandas`, `pandas`, and `*pandas*` (sphinx italics).

For documentation consistency, we should standardize these to one form. `pandas` is the preferable form in our documentation files.

Unfortunately I don't think we create a lint rule for this pattern since we have to be flexible to references and hyperlinks.

### PR DESCRIPTION
**Pull request by joybh98**
- [x] closes #32316 
- [x] tests added / passed
- [x] passes `black pandas`
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

As stated in the referred issue, _pandas_ reference in the docs should be standardised to pandas, not *pandas* or Pandas.

I've only changed the references in `doc/source/development/*.rst` for now, to see if the changes are okay or not

### CODE REVIEW

File: doc/source/development/code_style.rst [hunk 1]
```diff
@@ -21,7 +21,8 @@ Patterns
 foo.__class__
 -------------
 
-*pandas* uses 'type(foo)' instead 'foo.__class__' as it makes the code more
+
```
Comment (datapythonista) [inline doc/source/development/code_style.rst]: Can you delete this added blank line please.

File: doc/source/development/contributing.rst [hunk 2]
```diff
@@ -1272,9 +1272,9 @@ Running the performance test suite
 ----------------------------------
 
 Performance matters and it is worth considering whether your code has introduced
-performance regressions.  *pandas* is in the process of migrating to
+performance regressions.  pandas is in the process of migrating to
```
Comment (datapythonista) [inline doc/source/development/contributing.rst]: Do you mind leaving just one space before pandas

File: doc/source/development/contributing.rst [hunk 3]
```diff
@@ -1386,7 +1386,7 @@ Doing 'git status' again should give something like::
     #       modified:   /relative/path/to/file-you-added.py
     #
 
-Finally, commit your changes to your local repository with an explanatory message.  *Pandas*
+Finally, commit your changes to your local repository with an explanatory message.  Pandas
```
Comment (datapythonista) [inline doc/source/development/contributing.rst]: pandas in lower case, also, just leave one space before pandas.

File: doc/source/development/contributing.rst [hunk 4]
```diff
@@ -1434,7 +1434,7 @@ like::
     upstream        git://github.com/pandas-dev/pandas.git (fetch)
     upstream        git://github.com/pandas-dev/pandas.git (push)
 
-Now your code is on GitHub, but it is not yet a part of the *pandas* project.  For that to
+Now your code is on GitHub, but it is not yet a part of the pandas project.  For that to
```
Comment (datapythonista) [inline doc/source/development/contributing.rst]: one space only after the period

**Review by datapythonista — COMMENTED** [review]
Looks good.

What would be nice is to have a check in `ci/code_checks.sh` that greps for `Pandas` or `*pandas*` and fails if any occurrence is found. We do this for several patterns, you can use them as reference. Ideally we'd like to check at least all the `rst in `doc/` and `web/`.

**Comment by joybhallaa** [comment]
@datapythonista , sure I'll add the checks and update the required changes, I should add the checks in a separate PR right?

**Comment by datapythonista** [comment]
> @datapythonista , sure I'll add the checks and update the required changes, I should add the checks in a separate PR right?

You can add them here or in a separate PR, as you prefer. Not sure how many cases will still fail, if there are lots, probably better to open a new PR, so the PR is not so big.

**Comment by datapythonista** [comment]
@joybhallaa do you have time to add the checks and fix the CI?

**Comment by joybhallaa** [comment]
Yes!

File: doc/source/development/code_style.rst [hunk 5]
```diff
@@ -21,9 +21,9 @@ Patterns
 foo.__class__
 -------------
 
-*pandas* uses 'type(foo)' instead 'foo.__class__' as it makes the code more
-readable.
 
+pandas uses 'type(foo)' instead 'foo.__class__' as it is making the code more
```
Comment (datapythonista) [inline doc/source/development/code_style.rst]: next time please avoid side changes, like `makes` -> `is making` here.

**Review by datapythonista — APPROVED** [review]
Thanks, lgtm

**Review by WillAyd — APPROVED** [review]

**Comment by WillAyd** [comment]
Thanks @joybhallaa

**Pull request merged**